service/src/main/java/bio/terra/workspace/db/DbUtils.java

```diff
@@ -10,20 +13,33 @@
    * This method builds an SQL clause string for setting fields specified in the given parameters.
    * The method generates the column_name = :column_name list. It is an error if the params map is
    * empty.
+   *
+   * @param columnParams map of sql parameters.
+   * @param jsonColumns param columns that needs to be cast as jsonb.
    */
-  public static String setColumnsClause(MapSqlParameterSource columnParams) {
+  public static String setColumnsClause(MapSqlParameterSource columnParams, String... jsonColumns) {
     StringBuilder sb = new StringBuilder();
     String[] parameterNames = columnParams.getParameterNames();
-    if (parameterNames.length == 0) {
+    if (parameterNames.length == 0 && jsonColumns.length == 0) {
       throw new MissingRequiredFieldException("Must specify some data to be updated.");
     }
+    Set<String> jsonColumnSet = new HashSet(Arrays.asList(jsonColumns));
+    for (String jsonColumn : jsonColumns) {
+      sb.append(jsonColumn).append(" = cast(:").append(jsonColumn).append(" AS jsonb)");
+    }
     for (int i = 0; i < parameterNames.length; i++) {
       String columnName = parameterNames[i];
+      // If the param needs to be cast as jsonb, it has already been appended to sb so we can
+      // skip here.
+      if (jsonColumnSet.contains(columnName)) {
```

Comment:
I would have done
```
if (jsonColumnSet.contains(columnName) {
  sb.append(jsonColumn).append(" = cast(:").append(jsonColumn).append(" AS jsonb)");
} else {
  sb.append(columnName).append(" = :").append(columnName);
}
```
That is, the incoming columns are _possible_ json columns, but if they don't appear in the parameter source, then they have no effect. Then in the caller, you don't have to check if you are passing `attributes` and use different call forms.